### PR TITLE
fix(ci): run static analysis reporting only once per version COMPASS-7998

### PR DESCRIPTION
**Pull request by addaleax**

### CODE REVIEW

File: .evergreen.yml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-unset_function_vars: true
```
Comment (addaleax) [inline .evergreen.yml:1]: Drive-by: This field does not exist anymore

File: .evergreen/functions.yml [hunk 2]
```diff
@@ -331,7 +331,6 @@ functions:
         wait: true
     # copy ssh key (this key will be used to connect to ssh server)
     - command: shell.exec
-      silent: true
```
Comment (addaleax) [inline .evergreen/functions.yml:334]: Drive-by: `silent` would need to be set on the `params:`, not the command, if we wanted it to have an effect – but this should be fine either way, so let's just drop the invalid extra key

**Review by mabaasit — APPROVED** [review]

**Review by mcasimir — APPROVED** [review]
